media/js/src/AssetFilter.jsx

```diff
@@ -6,17 +6,80 @@ export default class AssetFilter extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            titleFilter: '',
             selectedOwner: 'all'
         };
-        this.handleChange = this.handleChange.bind(this);
     }
-    handleChange(e) {
-        console.log('handleChange');
+    handleTitleSearch(e) {
+        if (e.key === 'Enter') {
+            this.props.handleTitleFilterSearch();
+        }
+    }
+    handleOwnerChange(e) {
+        this.setState({selectedOwner: e.value});
     }
     render() {
         const { selectedOwner } = this.state;
 
+        let ownersOptions = [];
+        if (this.props.assets) {
```

Comment:
Another suggestion: If `assets` is always at least an empty array, it might be nice to use `reduce` here. That way you can skip using `if`, and use 'All Class Members' object as an initial value.

Reply:
Yeah... i made this in the most basic way possible, just using for loops. I'll use `reduce` instead.